backend/app/api/endpoints/rooms.py

```diff
@@ -2,11 +2,160 @@
 from starlette.responses import JSONResponse
 
 from app.api.deps import external_url_api_instance, verify_room_permissions
+from app.api.models import RoomConditions
+from app.core.app_state import app_state
+from app.core.models import GithubConditions
 from app.matrix.external_url import ExternalUrlAPI
 
 router = APIRouter()
 
 
+async def parse_github_org_data(github_conditions: GithubConditions):
+    orgs = github_conditions.orgs
+    owner_type = "org"
+    list_of_conditions = []
+    for org_name, org_data in orgs.items():
+        for repo_name, repo_data in org_data.repos.items():
+            repo_condition = RoomConditions(
+                type=owner_type,
+                third_party_account="Github",
+                owner={"parent": org_name, "child": repo_name},
+                condition_type="Repository",
+                data=repo_data,
+            )
+            list_of_conditions.append(repo_condition.dict())
```

Comment:
Are you converting all these models to dict here so you can pass it out through FastAPI's `JSONResponse`? I am kind of surprised that it can't handle being given a list of models and json'ing them itself.

Reply:
Yes, it is actually one of the annoying errors I've encountered so far. 
```
TypeError: Object of type RoomConditions is not JSON serializable
```

Reply:
There's a separate [`jsonable_encoder`](https://fastapi.tiangolo.com/tutorial/encoder/) available from FastAPI which we will have to import and parse pydantic models with it.

But I felt just converting it into dict might be easier.

Reply:
Curious